examples/async-tester/src/wasm.rs

```diff
@@ -1,3 +1,59 @@
+// This file is part of Gear.
+
+// Copyright (C) 2022-2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! This program demonstrates the use of [`msg`] syscalls from an async environment.
+//!
+//! `Handle` is async and gets a [`Kind`] in the payload, executing a syscall based on the `Kind`,
+//! and sending a message back to the source with the payload "PONG".
+//!
+//! [`Send`] uses the [`send_for_reply()`] syscall to send a message back to the source, containing
+//! the `kind` in the payload.
+//!
+//! [`SendWithGas(gas)`] uses the [`send_with_gas_for_reply()`] syscall to send a message back to
+//! the source, containing the `kind` in the payload, and `gas` as the gas limit.
+//!
+//! [`SendBytes`] uses the [`send_bytes_for_reply()`] syscall to send a message back to the source,
+//! containing the `kind` encoded as bytes in the payload.
+//!
+//! [`SendBytesWithGas(gas)`] uses the [`send_bytes_with_gas_for_reply()`] syscall to send a message
+//! back to the source, containing the `kind` encoded as bytes in the payload and `gas` as the gas
+//! limit.
+//!
+//! [`SendCommit`] uses the [`MessageHandle`], pushing the `kind` encoded as bytes and using
+//! [`commit_for_reply()`] to send the message back to the source.
+//!
+//! [`SendCommitWithGas(gas)`] uses the [`MessageHandle`], pushing the `kind` encoded as bytes and
+//! using [`commit_with_gas_for_reply()`] to send the message back to the source with `gas` as the
+//! gas limit.
```

Comment:
You don't need to literally explain every step of the program. These steps are clear, btw. State main of this tester crate (look at the usage).